devices/Vl53L0X/Vl53L0X.cs

```diff
@@ -0,0 +1,1241 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This code has been ported from the Dexter Industries Python code
+// https://github.com/DexterInd/DI_Sensors/blob/master/Python/di_sensors/VL53L0X.py
+// It is based as well on the offical ST Microelectronics API in C
+// https://www.st.com/content/st_com/en/products/embedded-software/proximity-sensors-software/stsw-img005.html
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Diagnostics;
+
+using System.IO;
+using System.Threading;
+
+namespace Iot.Device.Vl53L0X
+{
+    /// <summary>
+    /// Represents Vl53L0X
+    /// </summary>
+    public class Vl53L0X : IDisposable
+    {
+        /// <summary>
+        /// The default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x29;
+
+        private readonly bool _shouldDispose;
+        private readonly int _operationTimeout;
+
+        // Default address can be found in documentation
+        // page 18 with value 0x52 >> 1 = 0x29
+        private I2cDevice _i2cDevice;
+        private byte _stopData;
+        private uint _measurementTimingBudgetMicrosecond;
+        private bool _continuousInitialized = false;
+        private bool _highResolution;
+        private Precision _precision;
+
+        /// <summary>
+        /// Get the sensor information including internal signal and distance offsets
+        /// </summary>
+        public Information Information { get; private set; }
+
+        /// <summary>
+        /// Used to find a clean measurement when reading in single shot
+        /// </summary>
+        public int MaxTryReadSingle { get; set; }
+
+        /// <summary>
+        /// Set/Get high resolution measurement
+        /// </summary>
+        public bool HighResolution
+        {
+            get => _highResolution;
+            set
+            {
+                _highResolution = value;
+                WriteRegister((byte)Registers.SYSTEM_RANGE_CONFIG, _highResolution ? (byte)0x01 : (byte)0x00);
+            }
+        }
+
+        /// <summary>
+        /// Create a VL53L0X Sensor class
+        /// </summary>
+        /// <param name="i2cDevice">The I2C Device</param>
+        /// <param name="operationTimeoutMilliseconds">Timeout for reading data, by default 500 milliseonds</param>
+        /// <param name="shouldDispose">True to dispose the I2C Device at dispose</param>
+        public Vl53L0X(I2cDevice i2cDevice, int operationTimeoutMilliseconds = 500, bool shouldDispose = true)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+            _shouldDispose = shouldDispose;
+            _operationTimeout = operationTimeoutMilliseconds;
+            Reset();
+            Init();
+            GetInfo();
+            MaxTryReadSingle = 3;
+            // Set longer range
+            Precision = Precision.LongRange;
+#if NETCOREAPP2_1
```

Comment:
please get rid of the #if and adjust for the best. They've been introduced to support .NET 5, .NET Core 2.1 and .NET Standard